File: Java/Exercism/ResistorColorDuo.java
class ResistorColorDuo {
    int value(String[] colors) {
        
        String result = "";
        int iter = 0; 
        int maxIter = 2; //Two goes round

        for (String color : colors){
            if (iter < maxIter){
                switch(color){
                    case "black":
                        result += '0';
                        iter++;
                        break;
                    case "brown":
                        result += '1';
                        iter++;
                        break;
                    case "red":
                        result += '2';
                        iter++;
                        break;
                    case "orange":
                        result += '3';
                        iter++;
                        break;
                    case "yellow":
                        result += '4';
                        iter++;
                        break;
                    case "green":
                        result += '5';
                        iter++;
                        break;
                    case "blue":
                        result += '6';
                        iter++;
                        break;
                    case "violet":
                        result += '7';
                        iter++;
                        break;
                    case "grey":
                        result += '8';
                        iter++;
                        break;
                    case "white":
                        result += '9';
                        iter++;
                        break;
                }
            } else{ break; }
        }
        return Integer.parseInt(result); //Turn String to an Int
    }
}

File: JavaScript/Exercism/ResistorColorDuo.js
//
// This is only a SKELETON file for the 'Resistor Color Duo' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const decodedValue = colour => {

    let resistorColourDuoValue = '';
    
    for (let i = 0; i < Math.min(2, colour.length); i++)
    {
      switch (colour[i])
      {
        case 'black':
          resistorColourDuoValue += '0';
          break;
        case 'brown':
          resistorColourDuoValue += '1';
          break;
        case 'red':
          resistorColourDuoValue += '2';
          break;
        case 'orange':
          resistorColourDuoValue += '3';
          break;
        case 'yellow':
          resistorColourDuoValue += '4';
          break;
        case 'green':
          resistorColourDuoValue += '5';
          break;
        case 'blue':
          resistorColourDuoValue += '6';
          break;
        case 'violet':
          resistorColourDuoValue += '7';
          break;
        case 'grey':
          resistorColourDuoValue += '8';
          break;
        case 'white':
          resistorColourDuoValue += '9';
          break;
      }
    }
    return Number(resistorColourDuoValue);
  };
  